<h1>bedtools</h1>

<blockquote><p>A swiss-army knife of tools for genomic-analysis tasks.
Used to intersect, group, convert and count data in BAM, BED, GFF/GTF, VCF format.
More information: <a href="https://bedtools.readthedocs.io/en/latest/content/overview.html#summary-of-available-tools">https://bedtools.readthedocs.io/en/latest/content/overview.html#summary-of-available-tools</a>.</p></blockquote>

<ul>
<li>Intersect file [a] and file(s) [b] regarding the sequences' [s]trand and save the result to a specific file:</li>
</ul>


<p><code>bedtools intersect -a {{path/to/file_A}} -b {{path/to/file_B1 path/to/file_B2 ...}} -s &gt; {{path/to/output_file}}</code></p>

<ul>
<li>Intersect two files with a [l]eft [o]uter [j]oin, i.e. report each feature from <code>file1</code> and NULL if no overlap with <code>file2</code>:</li>
</ul>


<p><code>bedtools intersect -a {{path/to/file1}} -b {{path/to/file2}} -loj &gt; {{path/to/output_file}}</code></p>

<ul>
<li>Using more efficient algorithm to intersect two pre-sorted files:</li>
</ul>


<p><code>bedtools intersect -a {{path/to/file1}} -b {{path/to/file2}} -sorted &gt; {{path/to/output_file}}</code></p>

<ul>
<li>[g]roup a file based on the first three and the fifth [c]olumn and apply the sum [o]peration on the sixth column:</li>
</ul>


<p><code>bedtools groupby -i {{path/to/file}} -c 1-3,5 -g 6 -o sum</code></p>

<ul>
<li>Convert bam-formatted [i]nput file to a bed-formatted one:</li>
</ul>


<p><code>bedtools bamtobed -i {{path/to/file.bam}} &gt; {{path/to/file.bed}}</code></p>

<ul>
<li>Find for all features in <code>file1.bed</code> the closest one in <code>file2.bed</code> and write their [d]istance in an extra column (input files must be sorted):</li>
</ul>


<p><code>bedtools closest -a {{path/to/file1.bed}} -b {{path/to/file2.bed}} -d</code></p>
