<h1>nextclade</h1>

<blockquote><p>Bioinformatics tool for virus genome alignment, clade assignment and qc checks.
More information: <a href="https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-cli/reference.html">https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-cli/reference.html</a>.</p></blockquote>

<ul>
<li>Align sequences to user provided reference, outputting the alignment to a file:</li>
</ul>


<p><code>nextclade run {{path/to/sequences.fa}} {{[-r|--input-ref]}} {{path/to/reference.fa}} {{[-o|--output-fasta]}} {{path/to/alignment.fa}}</code></p>

<ul>
<li>Create a TSV report, auto-downloading the latest dataset:</li>
</ul>


<p><code>nextclade run {{path/to/fasta}} {{[-d|--dataset-name]}} {{dataset_name}} {{[-t|--output-tsv]}} {{path/to/report.tsv}}</code></p>

<ul>
<li>List all available datasets:</li>
</ul>


<p><code>nextclade dataset list</code></p>

<ul>
<li>Download the latest SARS-CoV-2 dataset:</li>
</ul>


<p><code>nextclade dataset get {{[-n|--name]}} sars-cov-2 {{[-o|--output-dir]}} {{path/to/directory}}</code></p>

<ul>
<li>Use a downloaded dataset, producing all outputs:</li>
</ul>


<p><code>nextclade run {{[-D|--input-dataset]}} {{path/to/dataset_dir}} {{[-O|--output-all]}} {{path/to/output_dir}} {{path/to/sequences.fasta}}</code></p>

<ul>
<li>Run on multiple files:</li>
</ul>


<p><code>nextclade run {{[-d|--dataset-name]}} {{dataset_name}} {{[-t|--output-tsv]}} {{path/to/output_tsv}} -- {{path/to/input_fasta_1 path/to/input_fasta_2 ...}}</code></p>

<ul>
<li>Try reverse complement if sequence does not align:</li>
</ul>


<p><code>nextclade run --retry-reverse-complement {{[-d|--dataset-name]}} {{dataset_name}} {{[-t|--output-tsv]}} {{path/to/output_tsv}} {{path/to/input_fasta}}</code></p>
