<h1>samtools</h1>

<blockquote><p>Tools for handling high-throughput sequencing (genomics) data.
Used for reading/writing/editing/indexing/viewing of data in SAM/BAM/CRAM format.
More information: <a href="https://www.htslib.org/doc/samtools.html">https://www.htslib.org/doc/samtools.html</a>.</p></blockquote>

<ul>
<li>Convert a SAM input file to BAM stream and save to file:</li>
</ul>


<p><code>samtools view -S {{[-b|--bam]}} {{input.sam}} &gt; {{output.bam}}</code></p>

<ul>
<li>Take input from <code>stdin</code> (-) and print the SAM header and any reads overlapping a specific region to <code>stdout</code>:</li>
</ul>


<p><code>{{other_command}} | samtools view {{[-h|--with-header]}} - chromosome:start-end</code></p>

<ul>
<li>Sort file and save to BAM (the output format is automatically determined from the output file&rsquo;s extension):</li>
</ul>


<p><code>samtools sort {{input}} {{[-o|--output]}} {{output.bam}}</code></p>

<ul>
<li>Index a sorted BAM file (creates <code>sorted_input.bam.bai</code>):</li>
</ul>


<p><code>samtools index {{sorted_input.bam}}</code></p>

<ul>
<li>Print alignment statistics about a file:</li>
</ul>


<p><code>samtools flagstat {{sorted_input}}</code></p>

<ul>
<li>Count alignments to each index (chromosome/contig):</li>
</ul>


<p><code>samtools idxstats {{sorted_indexed_input}}</code></p>

<ul>
<li>Merge multiple files:</li>
</ul>


<p><code>samtools merge {{output}} {{input1 input2 ...}}</code></p>

<ul>
<li>Split input file according to read groups:</li>
</ul>


<p><code>samtools split {{merged_input}}</code></p>
